tests/phpunit/tests/blocks/wpBlockPatternsRegistry.php

```diff
@@ -482,4 +482,80 @@ public function test_is_registered_for_known_pattern() {
 		$result = $this->registry->is_registered( 'test/one' );
 		$this->assertTrue( $result );
 	}
+
+	/**
+	 * Ensures theme patterns are registered on init.
+	 *
+	 * @ticket 59723
+	 *
+	 * @covers ::_register_theme_block_patterns
+	 */
+	public function test_register_theme_block_patterns_on_init() {
+		// This test needs to use access static class properties.
+		$registry = WP_Block_Patterns_Registry::get_instance();
```

Comment:
While most of the tests in this class can use the registry instance that is created during the `set_up()` method, these tests need to check he static instance stored to the class itself, as that is what gets affected during a normal WP application lifecycle. This also means we have to clean up after ourselves after registering patterns to that instance, otherwise we risk polluting other tests.